src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1709,175 +1703,74 @@ public static unsafe nuint WidenAsciiToUtf16(byte* pAsciiBuffer, char* pUtf16Buf
+            nuint finalOffsetWhereCanRunLoop = elementCount - (uint)Vector256<byte>.Count;
 
             do
             {
-                // In a loop, perform an unaligned read, widen to two vectors, then aligned write the two vectors.
+                Vector256<byte> asciiVector = Vector256.Load(pAsciiBuffer + currentOffset);
 
-                if (Sse2.IsSupported)
+                if (asciiVector.ExtractMostSignificantBits() != 0)
                 {
-                    asciiVector = Sse2.LoadVector128(pAsciiBuffer + currentOffset); // unaligned load
-                    containsNonAsciiBytes = (uint)Sse2.MoveMask(asciiVector) != 0;
-                }
-                else if (AdvSimd.Arm64.IsSupported)
-                {
-                    asciiVector = AdvSimd.LoadVector128(pAsciiBuffer + currentOffset);
-                    containsNonAsciiBytes = ContainsNonAsciiByte(asciiVector);
-                }
-                else
-                {
-                    throw new PlatformNotSupportedException();
-                }
-
-                if (containsNonAsciiBytes)
-                {
-                    // non-ASCII byte somewhere
-                    goto NonAsciiDataSeenInInnerLoop;
+                    break;
                 }
 
-                if (Sse2.IsSupported)
-                {
-                    Vector128<byte> low = Sse2.UnpackLow(asciiVector, zeroVector);
-                    Sse2.StoreAligned((byte*)pCurrentWriteAddress, low);
+                (Vector256<ushort> low, Vector256<ushort> upper) = Vector256.Widen(asciiVector);
+                low.Store(pCurrentWriteAddress);
+                upper.Store(pCurrentWriteAddress + Vector256<ushort>.Count);
 
-                    Vector128<byte> high = Sse2.UnpackHigh(asciiVector, zeroVector);
-                    Sse2.StoreAligned((byte*)pCurrentWriteAddress + SizeOfVector128, high);
-                }
-                else if (AdvSimd.Arm64.IsSupported)
-                {
-                    Vector128<ushort> low = AdvSimd.ZeroExtendWideningLower(asciiVector.GetLower());
-                    Vector128<ushort> high = AdvSimd.ZeroExtendWideningUpper(asciiVector);
-                    AdvSimd.Arm64.StorePair((ushort*)pCurrentWriteAddress, low, high);
-                }
-                else
-                {
-                    throw new PlatformNotSupportedException();
-                }
-
-                currentOffset += SizeOfVector128;
-                pCurrentWriteAddress += SizeOfVector128;
+                currentOffset += (nuint)Vector256<byte>.Count;
+                pCurrentWriteAddress += (nuint)Vector256<byte>.Count;
             } while (currentOffset <= finalOffsetWhereCanRunLoop);
 
-        Finish:
-
             return currentOffset;
-
-        NonAsciiDataSeenInInnerLoop:
-
-            // Can we at least widen the first part of the vector?
-
-            if (!containsNonAsciiBytes)
```

Comment:
I've removed this code, as it was impossible to satisfy this requirement as the jump was performed only when the flag was set to true:

https://github.com/dotnet/runtime/blob/85d638b76eb04409e5d5d96d0b811f85bce1002d/src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs#L1823-L1827